src/erasure-code/ErasureCodeInterface.h

```diff
@@ -370,6 +379,48 @@ namespace ceph {
     virtual int encode_chunks(const std::set<int> &want_to_encode,
                               std::map<int, bufferlist> *encoded) = 0;
 
+    /**
+     * Calculate the delta between the old_data and new_data
+     * buffers using xor, and returns the result in the delta buffer.
+     *
+     * Assumes old_data, new_data and delta are all buffers
+     * of the same length.
+     *
+     * @param [in] old_data first buffer to xor
+     * @param [in] new_data second buffer to xor
+     * @param [out] delta buffer to write the delta of old_data and new_data
+     */
+    virtual void encode_delta(const bufferptr &old_data,
```

Comment:
Do we have to use an 'out' param here (and below)?
Is it expected to contain any data beforehand (i.e. - is it an in-out param)?
If not - can it be created by the function and returned as its retval?